<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Valgrind::Internal::ValgrindConfigWidget</class>
 <widget class="QWidget" name="Valgrind::Internal::ValgrindConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>655</width>
    <height>364</height>
   </rect>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0" colspan="2">
    <widget class="QGroupBox" name="commonValgrindOptions">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Generic Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="valgrindExeLabel">
        <property name="text">
         <string>Valgrind executable:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="Utils::PathChooser" name="valgrindExeChooser" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="smcDetectionLabel">
        <property name="text">
         <string>Detect self-modifying code:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QComboBox" name="smcDetectionComboBox">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string>No</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Only on Stack</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Everywhere</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Everywhere Except in File-backend Mappings</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox" name="memcheckOptions_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Profiling Options</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="minimumInclusiveCostRatioLabel">
        <property name="toolTip">
         <string>Limits the amount of results the profiler gives you. A lower limit will likely increase performance.</string>
        </property>
        <property name="text">
         <string>Result view: Minimum event cost:</string>
        </property>
        <property name="buddy">
         <cstring>minimumInclusiveCostRatio</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="minimumInclusiveCostRatio">
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="maximum">
         <double>10.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QGroupBox" name="enableEventToolTips">
        <property name="title">
         <string>Show additional information for events in tooltips</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QCheckBox" name="enableCacheSim">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;
&lt;p&gt;Does full cache simulation.&lt;/p&gt;
&lt;p&gt;By default, only instruction read accesses will be counted (&quot;Ir&quot;).&lt;/p&gt;
&lt;p&gt;
With cache simulation, further event counters are enabled:
&lt;ul&gt;&lt;li&gt;Cache misses on instruction reads (&quot;I1mr&quot;/&quot;I2mr&quot;).&lt;/li&gt;
&lt;li&gt;Data read accesses (&quot;Dr&quot;) and related cache misses (&quot;D1mr&quot;/&quot;D2mr&quot;).&lt;/li&gt;
&lt;li&gt;Data write accesses (&quot;Dw&quot;) and related cache misses (&quot;D1mw&quot;/&quot;D2mw&quot;).&lt;/li&gt;&lt;/ul&gt;
&lt;/p&gt;

&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Enable cache simulation</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="enableBranchSim">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;
&lt;p&gt;Does branch prediction simulation.&lt;/p&gt;
&lt;p&gt;Further event counters are enabled: &lt;/p&gt;
&lt;ul&gt;&lt;li&gt;Number of executed conditional branches and related predictor misses (
&quot;Bc&quot;/&quot;Bcm&quot;).&lt;/li&gt;
&lt;li&gt;Executed indirect jumps and related misses of the jump address predictor (
&quot;Bi&quot;/&quot;Bim&quot;).&lt;/li&gt;&lt;/ul&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Enable branch prediction simulation</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="collectSystime">
           <property name="toolTip">
            <string>Collects information for system call times.</string>
           </property>
           <property name="text">
            <string>Collect system call time</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QCheckBox" name="collectBusEvents">
           <property name="toolTip">
            <string>Collect the number of global bus events that are executed. The event type &quot;Ge&quot; is used for these events.</string>
           </property>
           <property name="text">
            <string>Collect global bus events</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="visualisationMinimumInclusiveCostRatioLabel">
        <property name="text">
         <string>Visualization: Minimum event cost:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="visualisationMinimumInclusiveCostRatio">
        <property name="prefix">
         <string/>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>50.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>1</width>
       <height>500</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="0">
    <widget class="QGroupBox" name="memcheckOptions">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Memory Analysis Options</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0" colspan="2">
       <widget class="QCheckBox" name="trackOrigins">
        <property name="text">
         <string>Track origins of uninitialized memory</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QCheckBox" name="showReachable">
        <property name="text">
         <string>Show reachable and indirectly lost blocks</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="leakCheckOnFinishLabel">
        <property name="text">
         <string>Check for leaks on finish:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QComboBox" name="leakCheckOnFinish">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>No</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Summary Only</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Full</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="numCallersLabel">
        <property name="text">
         <string>Backtrace frame count:</string>
        </property>
        <property name="buddy">
         <cstring>numCallers</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QSpinBox" name="numCallers">
          <property name="minimum">
           <number>5</number>
          </property>
          <property name="maximum">
           <number>50</number>
          </property>
          <property name="value">
           <number>12</number>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Suppression files:</string>
        </property>
        <property name="buddy">
         <cstring>suppressionList</cstring>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QListView" name="suppressionList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,1">
          <property name="sizeConstraint">
           <enum>QLayout::SetMinimumSize</enum>
          </property>
          <item>
           <widget class="QPushButton" name="addSuppression">
            <property name="text">
             <string>Add...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="removeSuppression">
            <property name="text">
             <string>Remove</string>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
